src/gtk/webview_webkit2.cpp

```diff
@@ -1093,13 +1099,81 @@ wxString wxWebViewWebKit::GetPageText() const
     return wxString();
 }
 
-void wxWebViewWebKit::RunScript(const wxString& javascript)
+static void wxgtk_run_javascript_cb(WebKitWebView *,
+                                    GAsyncResult *res,
+                                    GAsyncResult **res_out)
 {
+    g_object_ref(res);
+    *res_out = res;
+}
+
+bool JSResultToString(GObject *object, GAsyncResult *result, wxString* output)
+{
+    wxGtkError error;
+    wxWebKitJavascriptResult js_result(webkit_web_view_run_javascript_finish(WEBKIT_WEB_VIEW (object),
+        (GAsyncResult *)result, error.Out()));
+
+    if ( !&*js_result )
```

Comment:
This looks very confusing and is undefined in C++ (you can't dereference a null pointer). Add a `IsOk()` method to `wxWebKitJavascriptResult` instead, or just write `if ( !js_result )` which should already work via implicit conversion.